pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -604,6 +632,64 @@ func (m *kubeGenericRuntimeManager) computePodActions(pod *v1.Pod, podStatus *ku
 			// If the container failed the startup probe, we should kill it.
 			message = fmt.Sprintf("Container %s failed startup probe", container.Name)
 		} else {
+			if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+				keepCount++
+				if container.Resources.Limits == nil || len(pod.Status.ContainerStatuses) == 0 {
+					continue
+				}
+				apiContainerStatus, exists := podutil.GetContainerStatus(pod.Status.ContainerStatuses, container.Name)
+				if !exists || apiContainerStatus.State.Running == nil ||
+					containerStatus.State != kubecontainer.ContainerStateRunning ||
+					containerStatus.ID.String() != apiContainerStatus.ContainerID ||
```

Comment:
Why does the ID need to be the same?  Is this protecting against resizes during container restarts or something?

Reply:
Yes. I ran into this with RestartContainer policy where we would enter the resize operation code while the container is already being restarted, and it would get picked for spurious/duplicate kill+starts.

Reply:
A comment explaining this would be helpful.

Reply:
Added comments. I also changed the entry condition from Resources.Limits to Resources as we need to execute this when only CPU request changes. Please take a look.

Reply:
looks good